### PR TITLE
getallcolumn refactor changes as per PR

### PR DESCRIPTION
**Pull request by steadtler**

### CODE REVIEW

File: sql/sql_expression.h [hunk 1]
```diff
@@ -504,23 +504,31 @@ struct GetAllColumnsOutput {
...
 
-    ColumnFilter(Exec exec)
-        : exec(std::move(exec))
+    ColumnFilter(Exec exec_)
+        : exec(std::move(exec_))
     {
+        init = true;
+    }
+
+    static ColumnFilter identity() {
```
Comment (guyd) [inline sql/sql_expression.h]: I was thinking of something like that:
.h
`static ColumnFilter identity`
.cc
```
ColumnFilter::identity = ColumnFilter(
[](const ColumnName & columnName) {
   return columnName;
});
```
Everywhere else
`ColumnFilter::identity`
And I would have removed the unnecessary default constructor and the init member.

**Comment by guyd** [comment]
Minor comment which is more a manner of taste.
+1

File: sql/sql_expression.h (hunk 1 above)
Comment (steadtler) [inline sql/sql_expression.h]: But one of the reason for this refactor is that I want to know when the filter does nothing...

Comment (guyd) [inline sql/sql_expression.h]: Not sure I understand how one way or the other makes it easier to know that the filter does nothing.

**Pull request closed without merging**